tests/Utils/ClearStateTestListener.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace SimpleSAML\Test\Utils;
+
+use PHPUnit_Framework_Test;
+use PHPUnit_Framework_TestSuite;
+
+/**
+ * A PHPUnit test listener that attempts to clear global state and cached SSP configuration between test run
+ */
+class ClearStateTestListener extends \PHPUnit_Framework_BaseTestListener
```

Comment:
In our earlier conversation (https://groups.google.com/forum/#!topic/simplesamlphp-dev/OKfZtXrc3P0) I thought you didn't want to go with the base class because it required more knowledge on the part of the test writer on how the SSP project does testing.

I have no real preference between the approaches. The test listener did not add meaningfully to the test execution time. I think the subclass approach would give a test implementation more control over what they wanted done with the state, and is easier for people to understand.